src/plugins/process/process.c

```diff
@@ -0,0 +1,286 @@
+	if (pp == NULL)
+	{
+		// process initialization
+		Process * process = elektraMalloc (sizeof (Process));
+		KeySet * processConfig = elektraPluginGetConfig (handle);
+		process->pluginName = ksLookupByName (processConfig, "/plugin", KDB_O_POP);
+		process->pluginConfig = ksDup (processConfig);
+		ksAppendKey (processConfig, process->pluginName);
+		process->plugin = NULL;
+
+		if ((pp = elektraPluginProcessInit (errorKey)) == NULL) return ELEKTRA_PLUGIN_STATUS_ERROR;
+
+		elektraPluginProcessSetData (pp, process);
+		elektraPluginSetData (handle, pp);
+		if (!elektraPluginProcessIsParent (pp)) elektraPluginProcessStart (handle, pp);
+	}
+
+	if (elektraPluginProcessIsParent (pp)) return elektraPluginProcessOpen (pp, errorKey);
+
+	Process * process = elektraPluginProcessGetData (pp);
+
+	// First time child initialization
+	// elektraInvokeOpen will call the plugin's open function, this has to happen in the other process
+
+	if (process->plugin == NULL && !isContractKey (errorKey) && validPluginName (process->pluginName, errorKey))
+	{
+		process->plugin = elektraInvokeOpen (keyString (process->pluginName), process->pluginConfig, errorKey);
+		if (!process->plugin)
+		{
+			ELEKTRA_SET_ERRORF (197, errorKey, "Failed to open the proxied plugin %s", keyString (process->pluginName));
+			return ELEKTRA_PLUGIN_STATUS_ERROR;
+		}
+		return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+	}
+
+	if (process->plugin == NULL) return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+
+	// Otherwise just call the open function if it exists and don't reinitialize with elektraInvokeOpen
+	int ret = elektraInvoke1Arg (process->plugin, "open", errorKey);
+	if (ret == -2) return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+	return ret;
+}
+
+int elektraProcessClose (Plugin * handle, Key * errorKey)
+{
+	ElektraPluginProcess * pp = elektraPluginGetData (handle);
+	if (!pp) return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+	Process * process = elektraPluginProcessGetData (pp);
+	if (elektraPluginProcessIsParent (pp))
+	{
+		ElektraPluginProcessCloseResult result = elektraPluginProcessClose (pp, errorKey);
+		if (result.cleanedUp)
+		{
+			cleanup (process, errorKey);
+			elektraPluginSetData (handle, NULL);
+		}
+		return result.result;
+	}
+
+
+	// Handle may be null if initialization failed in the child, ignore that it will exit anyway afterwards
+	if (process->plugin == NULL) return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+
+	int ret = elektraInvoke1Arg (process->plugin, "close", errorKey);
+	if (ret == -2) return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+}
+
+static void adjustContract (KeySet * pluginContract, KeySet * contract)
+{
+	ksRewind (pluginContract);
+	Key * cur;
+	while ((cur = ksNext (pluginContract)) != NULL)
+	{
+		Key * cpy = keyDup (cur);
+		keySetBaseName (cpy, NULL);
+		if (!elektraStrCmp ("infos", keyBaseName (cpy)))
+		{
+			keySetBaseName (cpy, NULL);
+			keySetBaseName (cpy, NULL);
+			keyAddBaseName (cpy, "process");
```

Comment:
Why to process and not pluginName?

Reply:
the contract returned from this plugin seems to be expected to be called process, i don't recall the error but i think it was something like contract missing.

Reply:
Ok. Did you try to use `process#1 plugin=dump process#2 plugin=other`? Iirc this should allow different contracts for plugins with the same name.

Reply:
unfortunately this still doesn't resolve the issue of not knowing which plugin shall be proxied upon getting the contract, so it is only of little use currently.

Reply:
Ok, please resolve the other small things so that we can get this into master.